core/src/main/java/org/wildfly/channel/version/VersionMatcher.java

```diff
@@ -27,6 +27,13 @@
  *
  * The {@code #COMPARATOR} provides the implementation to compare the versions.
  * The latest one is the "last" determine by this comparator.
+ *
+ * This comparator will split the versions to determine their digits and compare them numerically.
```

Comment:
What does the comparator split version on? What are the spliting characters?
Looking at implementation of `VersionMatcher#COMPARATOR`, I don't think any spliting is going on. 

It seems to expect version in the format of (epoch:)versionstring, where 
- epoch is optional but must be numeric if present. If epoch is missing, it is considered 0.
- version is alphanumerical

Working left to right, it will
- evaluate epoch as a number (epoch defaults to 0 if not present). If epochs are equal, continue to next symbol
- compare character to character. If they are equal, continue to next symbol.
- read numbers as blocks and compare them, ie. 002 > 1. If equal, continue to next symbol.
- if one version is a "head" of the other, ie. "FOO" and "FOOBAR", the longer version is considered as newer version

Reply:
The fact that colon is a special character is important here, as parsing version "1.0.0:Alpha" will fail with NumberFormatException but parsing "1.0.0-Alpha" will not. This is not documented.

Reply:
I've update the PR

Reply:
I think it is much more correct and readable now.